pkg/grpc/identity/identity.go

```diff
@@ -0,0 +1,27 @@
+// Package identity contains protobuf types for identity management.
+package identity
+
+import (
+	"crypto/sha256"
+
+	"google.golang.org/protobuf/proto"
+
+	"github.com/pomerium/pomerium/pkg/encoding/base58"
+)
+
+// Clone clones the Provider.
+func (x *Provider) Clone() *Provider {
+	return proto.Clone(x).(*Provider)
+}
+
+// Hash computes a sha256 hash of the provider's fields. It excludes the Id field.
```

Comment:
Why is it important to exclude the Id field?

Reply:
Because the ID is the hash. If it were included re-computing the Hash would result in the ID changing every time.